common/djangoapps/track/tests/test_contexts.py

```diff
@@ -1,23 +1,35 @@
 # pylint: disable=missing-docstring
 
+import ddt
 from unittest import TestCase
 
 from track import contexts
 
 
+@ddt.ddt
 class TestContexts(TestCase):
 
     COURSE_ID = 'test/course_name/course_run'
+    SPLIT_COURSE_ID = 'course-v1:test+course_name+course_run'
     ORG_ID = 'test'
 
-    def test_course_id_from_url(self):
-        self.assert_parses_course_id_from_url('http://foo.bar.com/courses/{course_id}/more/stuff')
+    @ddt.data(
+        ('http://foo.bar.com/courses/{course_id}', COURSE_ID),
+        ('http://foo.bar.com/courses/{course_id}/more/stuff', COURSE_ID),
+        ('http://foo.bar.com/courses/{course_id}?format=json', COURSE_ID),
+        ('http://foo.bar.com/courses/{course_id}', SPLIT_COURSE_ID),
+        ('http://foo.bar.com/courses/{course_id}/more/stuff', SPLIT_COURSE_ID),
+        ('http://foo.bar.com/courses/{course_id}?format=json', SPLIT_COURSE_ID)
+    )
```

Comment:
We have occasionally written tests like this using `ddt.unpack` and `itertools.product`, so that you don't have to manually least out multiple copies of the url context.